test/extended/dns/dns.go

```diff
@@ -31,8 +32,13 @@ import (
 	"github.com/openshift/origin/test/extended/util/image"
 )
 
-func createDNSPod(namespace, probeCmd string) *kapiv1.Pod {
-	nodeSelector := map[string]string{"node-role.kubernetes.io/master": ""}
+func createDNSPod(oc *exutil.CLI, namespace, probeCmd string) *kapiv1.Pod {
+	controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	nodeSelector := map[string]string{}
+	if *controlPlaneTopology != ocpv1.ExternalTopologyMode {
+		nodeSelector = map[string]string{"node-role.kubernetes.io/master": ""}
+	}
```

Comment:
Instead of running an `oc` command everytime in `createDNSPod`, it would be better to set the `nodeSelector` once in __ and pass in the nodeSelector as a parameter instead of oc as a parameter.  E.g. move this code block to L417 in `Describe("[sig-network-edge] DNS", func() {`.